packages/flutter_tools/lib/src/device.dart

```diff
@@ -175,11 +175,27 @@ class DeviceManager {
   Future<List<Device>> findTargetDevices(FlutterProject flutterProject) async {
```

Comment:
The method comment needs updating; the part about only filtering if --all is not specified is out of date.